### PR TITLE
Initial documentation of release process

### PR DESCRIPTION
**Pull request by kevmoo**
Fixes https://github.com/GoogleCloudPlatform/functions-framework-dart/issues/82

### CODE REVIEW

File: docs/release_process.md [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+It is hard to be _too_ careful here.
+
+Assuming that we want to release both
+[`functions_framework`](https://pub.dev/packages/functions_framework) and
+[`functions_framework_builder`](https://pub.dev/packages/functions_framework_builder)
+at the same time, follow these steps.
+
+1. Prepare a PR to release functions_framework.
+   - Remove the `-dev` from `pubspec.yaml` and `CHANGELOG.md`.
+2. Get the PR approved and commit it.
```
Comment (grant) [inline docs/release_process.md:23]: Use "1." Everywhere for ordered lists in markdown

File: docs/release_process.md [hunk 2]
```diff
@@ -0,0 +1,36 @@
...
+before a package is published. The Dart team has spent hours and in some cases
+days fixing problems due to a mistake when publishing.
+It is hard to be _too_ careful here.
+
+Assuming that we want to release both
+[`functions_framework`](https://pub.dev/packages/functions_framework) and
+[`functions_framework_builder`](https://pub.dev/packages/functions_framework_builder)
+at the same time, follow these steps.
+
+1. Prepare a PR to release functions_framework.
```
Comment (grant) [inline docs/release_process.md:21]: Ticks for `functions_framework`?

File: docs/release_process.md [hunk 3]
```diff
@@ -0,0 +1,36 @@
...
+   - Remove the `-dev` from `pubspec.yaml` and `CHANGELOG.md`.
+2. Get the PR approved and commit it.
+3. Publish `functions_framework`.
+4. Tag the commit with the package name and version.
```
Comment (grant) [inline docs/release_process.md:25]: Example tag?

**Review by grant — APPROVED** [review]
Didn't give a thorough review.

I assume these steps were tried out verbatim?

**Comment by kevmoo** [comment]
> I assume these steps were tried out verbatim?

I have followed these steps many times for many such projects in the past.

We will "dry run" these instructions with `0.3.0`

File: docs/release_process.md (hunk 2 above)
Comment (kevmoo) [reply docs/release_process.md:21]: ack

File: docs/release_process.md (hunk 1 above)
Comment (kevmoo) [reply docs/release_process.md:23]: ack

**Review by subfuzion — APPROVED** [review]
LGTM

File: docs/release_process.md (hunk 3 above)
Comment (kevmoo) [reply docs/release_process.md:25]: ack

**Comment by kevmoo** [comment]
I'll need another approval @grant @subfuzion – updating a PR clears out reviews

**Review by subfuzion — APPROVED** [review]

**Review by subfuzion — APPROVED** [review]
👍

**Pull request merged**